### PR TITLE
Include Platform Handle in surface array

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Adds a platform surface handle to the surface array.  This enables custom skia gpu to access window info needed for creating rendertarget for that window. 

## How was the solution implemented (if it's not obvious)?
For Android, the native window is now accessible through IPlatform.Handle. on Windows, Linux,  and Android, the platform handle is added.
## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019387-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019390-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/WindowImpl.cs [hunk 1]
```diff
@@ -459,7 +459,7 @@ private WindowTransparencyLevel Win10EnableBlur(WindowTransparencyLevel transpar
             }
         }
 
-        public IEnumerable<object> Surfaces => new object[] { Handle, _gl, _framebuffer };
+        public IEnumerable<object> Surfaces => new object[] { Handle, _gl, _framebuffer};
```
Comment (kekekeks) [inline src/Windows/Avalonia.Win32/WindowImpl.cs:462]: WindowImplPlatformHandle is not actually exposed here

File: src/Avalonia.X11/X11Window.cs [hunk 2]
```diff
@@ -1142,5 +1144,25 @@ public void SetWindowManagerAddShadowHint(bool enabled)
...
+
+        public class SurfacePlatformHandle : IPlatformNativeSurfaceHandle
+        {
+            private readonly X11Window _owner;
+
+            public PixelSize Size => _owner.ToPixelSize(_owner.ClientSize);
+
+            public double Scaling => _owner.RenderScaling;
+
+            public SurfacePlatformHandle(IntPtr handle, string? descriptor, X11Window owner)
```
Comment (kekekeks) [inline src/Avalonia.X11/X11Window.cs:1157]: I think the way you've done it for Win32 is better. So just pass X11Window

File: src/Avalonia.X11/X11Window.cs [hunk 3]
```diff
@@ -135,7 +135,7 @@ public X11Window(AvaloniaX11Platform platform, IWindowImpl popupParent)
             else
                 _renderHandle = _handle;
                 
-            Handle = new PlatformHandle(_handle, "XID");
+            Handle = new SurfacePlatformHandle(_handle, "XID", this);
```
Comment (kekekeks) [inline src/Avalonia.X11/X11Window.cs:138]: You need to use `_renderHandle`. In some cases we set up a child window to render into.

**Review by kekekeks — CHANGES_REQUESTED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019454-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/WindowImpl.cs (hunk 1 above)
Comment (emmauss) [reply src/Windows/Avalonia.Win32/WindowImpl.cs]: Like this, `public IEnumerable<object> Surfaces => new object[] { (IPlatformNativeSurfaceHandle)Handle, _gl, _framebuffer };`.
The Handle is still an object of type `IPlatformNativeSurfaceHandle`.

File: src/Avalonia.X11/X11Window.cs (hunk 2 above)
Comment (emmauss) [reply src/Avalonia.X11/X11Window.cs]: Addressed

File: src/Avalonia.X11/X11Window.cs (hunk 3 above)
Comment (emmauss) [reply src/Avalonia.X11/X11Window.cs]: Addressed

**Review by kekekeks — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0019461-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**